src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs

```diff
@@ -68,6 +70,55 @@ public void CompleteStream()
 
     private static readonly byte[] TestData = Encoding.UTF8.GetBytes("Hello world");
 
+    [ConditionalFact]
+    [MsQuicSupported]
+    public async Task GET_Metrics_HttpProtocolAndTlsSet()
+    {
+        // Arrange
+        var builder = CreateHostBuilder(context => Task.CompletedTask);
+
+        using (var host = builder.Build())
+        {
+            var meterFactory = host.Services.GetRequiredService<IMeterFactory>();
+
+            var tcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+            using var connectionDuration = new InstrumentRecorder<double>(meterFactory, "Microsoft.AspNetCore.Server.Kestrel", "connection-duration");
+            using var measurementReporter = new MeasurementReporter<double>(meterFactory, "Microsoft.AspNetCore.Server.Kestrel", "connection-duration");
+            measurementReporter.Register(m =>
+            {
+                tcs.SetResult();
+            });
+
+            await host.StartAsync();
+            var client = HttpHelpers.CreateClient();
+
+            // Act
+            var request1 = new HttpRequestMessage(HttpMethod.Get, $"https://127.0.0.1:{host.GetPort()}/");
+            request1.Version = HttpVersion.Version30;
+            request1.VersionPolicy = HttpVersionPolicy.RequestVersionExact;
+
+            var response1 = await client.SendAsync(request1, CancellationToken.None);
+            response1.EnsureSuccessStatusCode();
+
+            // Dispose the client to end the connection.
+            client.Dispose();
+            // Wait for measurement to be available.
+            await tcs.Task.DefaultTimeout();
+
+            // Assert
+            Assert.Collection(connectionDuration.GetMeasurements(),
+                m =>
+                {
+                    Assert.True(m.Value > 0);
+                    Assert.Equal("Tls13", m.Tags.ToArray().Single(t => t.Key == "tls-protocol").Value);
```

Comment:
Missing tls-protocol tests for http/1.1 and http/2

Reply:
I added a test for HTTP/2. The logic is the same for HTTP/1.1.